auto_submit/lib/service/validation_service.dart

```diff
@@ -128,40 +143,148 @@ class ValidationService {
     for (ValidationResult result in results) {
       if (!result.result && result.action == Action.REMOVE_LABEL) {
         final String commmentMessage = result.message.isEmpty ? 'Validations Fail.' : result.message;
-        await gitHubService.createComment(slug, prNumber, commmentMessage);
-        await gitHubService.removeLabel(slug, prNumber, Config.kAutosubmitLabel);
-        log.info('auto label is removed for ${slug.fullName}, pr: $prNumber, due to $commmentMessage');
+
+        String message = 'auto label is removed for ${slug.fullName}, pr: $prNumber, due to $commmentMessage';
+
+        log.info(message);
+        await removeLabelAndComment(
+            githubService: gitHubService,
+            repositorySlug: slug,
+            prNumber: prNumber,
+            prLabel: Config.kAutosubmitLabel,
+            message: message);
+
         shouldReturn = true;
       }
     }
+
     if (shouldReturn) {
       log.info('The pr ${slug.fullName}/$prNumber with message: $ackId should be acknowledged.');
       await pubsub.acknowledge('auto-submit-queue-sub', ackId);
       log.info('The pr ${slug.fullName}/$prNumber is not feasible for merge and message: $ackId is acknowledged.');
       return;
     }
+
     // If PR has some failures to ignore temporarily do nothing and continue.
     for (ValidationResult result in results) {
       if (!result.result && result.action == Action.IGNORE_TEMPORARILY) {
         return;
       }
     }
+
     // If we got to this point it means we are ready to submit the PR.
-    bool processed = await processMerge(config, result, messagePullRequest);
-    if (processed) await pubsub.acknowledge('auto-submit-queue-sub', ackId);
+    bool processed = await processMerge(config: config, queryResult: result, messagePullRequest: messagePullRequest);
+
+    if (!processed) {
+      String message = 'auto label is removed for ${slug.fullName}, pr: $prNumber, merge did not succeed.';
```

Comment:
It would be helpful if we can post the real error `message` to the PR comment.

Reply:
Done!